Tickets/dm 25789

jeffcarlin opened the pull request.

bechtol reviewed (commented):
Looks good. This is a big simplification, and should in the future make it easier to modify to subtract off the statistical shot noise. Three more general comments:

(1) It would be helpful to see a plot of RMS vs. magnitude to get some intuition for how the new algorithm works. Suggest to use a lower SNR threshold to make the visualization, e.g., SNR>50 to see the variation in RMS vs. flux
(2) It seems that `cal_RMS` function is not needed, as that calculation can be done with `np.std` or `np.nanstd`?
(3) It sounded like we converged on removing the big blocks of code that have been commented out from the old algorithm.

jeffcarlin commented on python/lsst/faro/utils/phot_repeat.py:
```diff
@@ -1,29 +1,48 @@
         return {'nomeas': np.nan*u.mmag}
 
 
+def cal_RMS(x):
```
Good point. At some point I thought I had decided it was not the same, but I was mistaken. I've changed it to `np.nanstd` (I think it's good to go ahead and ditch the NaNs so the entire metric measurement doesn't become a NaN.)

jeffcarlin commented on python/lsst/faro/utils/phot_repeat.py:
```diff
@@ -1,29 +1,48 @@
+
+
+def calcPhotRepeat(matches, magKey):
+    matches_rms = matches.aggregate(cal_RMS, field=magKey)*1000.0*u.mmag
```
Nice - that's much better doing everything in terms of the units.

jeffcarlin commented on python/lsst/faro/utils/phot_repeat.py:
```diff
@@ -1,29 +1,48 @@
+    matches_rms = matches.aggregate(cal_RMS, field=magKey)*1000.0*u.mmag
+    matches_count = matches.aggregate(np.count_nonzero, field=magKey)
+    matches_mean = matches.aggregate(np.mean, field=magKey)*u.mag
+    magDiff = []
```
Yep - implemented.

jeffcarlin commented on python/lsst/faro/utils/phot_repeat.py:
```diff
@@ -1,29 +1,48 @@
-    nMinPhotRepeat = 50
     if filteredCat.count > nMinPhotRepeat:
-        phot_resid_meas = calcPhotRepeat(filteredCat, magKey, numRandomShuffles=50, randomSeed=randomSeed)
+        phot_resid_meas = calcPhotRepeat(filteredCat, magKey)
```
I like that idea. I added a couple lines to do that check.

bechtol commented:
The updates look good. One other thought is to add doc strings for the new `photRepeat` and `calcPhotRepeat` functions.

jeffcarlin commented:
Yep - I was thinking the same re: docstrings. I’m running it on all visits from RC2 to test the persistence of extras. Will add docstrings before I “finalize”.


On Apr 6, 2021, at 7:04 PM, Keith Bechtol ***@***.******@***.***>> wrote:



The updates look good. One other thought is to add doc strings for the new photRepeat and calcPhotRepeat functions.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/lsst/faro/pull/78#issuecomment-814547068>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFDBY7M7PP2TO47W4H7U7Y3THO4RNANCNFSM42NUOTAA>.

**********************************************
Jeff Carlin

DM Science Validation Scientist and Community Scientist
Rubin Observatory/Legacy Survey of Space and Time (LSST)
950 N. Cherry Ave.
Tucson, AZ 85719
***@***.******@***.***>
(520) 318-8533

jeffcarlin commented:
I believe I have addressed all comments, and have added docstrings. I'm a novice at writing docstrings, so please let me know if anything obvious is amiss.

bechtol commented on python/lsst/faro/measurement/MatchedCatalogMeasurementTasks.py:
```diff
@@ -38,96 +48,110 @@ def __init__(self, config: PA1TaskConfig, *args, **kwargs):
+        Returns
+        -------
+        measurement : `lsst.verify.Measurement`
+        Measurement of the percentage of repeatability outliers, and associated metadata.
```
It looks like this line would be indented to match convention

bechtol commented on python/lsst/faro/measurement/MatchedCatalogMeasurementTasks.py:
```diff
@@ -38,96 +48,110 @@ def __init__(self, config: PA1TaskConfig, *args, **kwargs):
+        Returns
+        -------
+        measurement : `lsst.verify.Measurement`
+        Measurement of the repeatability and its associated metadata.
```
It looks like this line would be indented to match convention
